### PR TITLE
Format `class` definitions

### PR DESCRIPTION
**Pull request by MichaReiser**
## Summary

This PR adds basic formatting for `class` definitions.

## Test Plan

Ran the stability check to verify that the class definition formatting does not introduce any new instabilities. 
Added new snapshot tests.

### CODE REVIEW

**Comment by MichaReiser** [comment]
Current dependencies on/for this PR:
* main
  * **PR #5289** <a href="https://app.graphite.dev/github/pr/astral-sh/ruff/5289" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/astral-sh/ruff/5289?utm_source=stack-comment).

File: Cargo.toml [hunk 1]
```diff
@@ -50,15 +50,15 @@ toml = { version = "0.7.2" }
 # v0.0.1
 libcst = { git = "https://github.com/charliermarsh/LibCST", rev = "80e4c1399f95e5beb532fdd1e209ad2dbb470438" }
 # v0.0.3
-ruff_text_size = { git = "https://github.com/astral-sh/RustPython-Parser.git", rev = "08ebbe40d7776cac6e3ba66277d435056f2b8dca" }
+ruff_text_size = { git = "https://github.com/astral-sh/RustPython-Parser.git", rev = "f60e204b73b95bdb6ce87ccd0de34081b4a17c11" }
```
Comment (MichaReiser) [inline Cargo.toml:53]: Pulls in https://github.com/astral-sh/RustPython-Parser/pull/14

File: crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__black_test__class_blank_parentheses_py.snap
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/snapshots/ruff_python_formatter__tests__black_test__class_blank_parentheses_py.snap:1]: Jepee

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs [hunk 2]
```diff
@@ -1,12 +1,113 @@
...
+impl Format<PyFormatContext<'_>> for FormatInheritanceClause<'_> {
+    fn fmt(&self, f: &mut Formatter<PyFormatContext<'_>>) -> FormatResult<()> {
+        let StmtClassDef {
+            bases,
+            keywords,
+            name,
+            ..
+        } = self.class_definition;
+
+        let separator = format_with(|f| write!(f, [text(","), soft_line_break_or_space()]));
```
Comment (MichaReiser) [inline crates/ruff_python_formatter/src/statement/stmt_class_def.rs:63]: We should probably extract a helper at some point that joins a list with `,` and `soft_line_break_or_space` and preserves any trailing magic comma.

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs [hunk 3]
```diff
@@ -1,12 +1,133 @@
...
+        let source = f.context().contents();
+
+        let mut joiner = f.join_with(&separator);
+
+        if let [first, rest @ ..] = bases.as_slice() {
```
Comment (konstin) [inline crates/ruff_python_formatter/src/statement/stmt_class_def.rs:88]: nit: `split_first`

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs [hunk 4]
```diff
@@ -1,12 +1,133 @@
...
+            // Manually handle parentheses here because the expression is parenthesized incorrectly assumes that the
+            // parentheses from the inheritance clause belong to the expression.
```
Comment (konstin) [inline crates/ruff_python_formatter/src/statement/stmt_class_def.rs:90]: don't understand that comment

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs [hunk 5]
```diff
@@ -1,12 +1,133 @@
...
+        // handled in fmt_fields
+        Ok(())
+    }
+}
+
+struct FormatInheritanceClause<'a> {
+    class_definition: &'a StmtClassDef,
+}
+
+impl Format<PyFormatContext<'_>> for FormatInheritanceClause<'_> {
```
Comment (konstin) [inline crates/ruff_python_formatter/src/statement/stmt_class_def.rs:74]: Is this effectively formatting a function call?

**Review by konstin — APPROVED** [review]

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs (hunk 5 above)
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/statement/stmt_class_def.rs:74]: It's similar but it doesn't need to support any of the complexity like chaining.

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs (hunk 3 above)
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/statement/stmt_class_def.rs:88]: Neat

File: crates/ruff_python_formatter/src/statement/stmt_class_def.rs (hunk 4 above)
Comment (MichaReiser) [reply crates/ruff_python_formatter/src/statement/stmt_class_def.rs:90]: I reworded the comment and added an example.

**Pull request merged**